graph/unionfind.go

```diff
@@ -0,0 +1,76 @@
+// Union Find Algorithm or Dynamic Connectivity algorithm, often implemented with the help 
+
+//of the union find data structure,
+
+// is used to efficiently maintain connected components in a graph that undergoes dynamic changes,
+
+// such as edges being added or removed over time
+
+// Worst Case Time Complexity: The time complexity of find operation is nearly constant or 
+
+//O(α(n)), where where α(n) is the inverse Ackermann function
+
+// practically, this is a very slowly growing function making the time complexity for find 
+
+//operation nearly constant.
+
+// The time complexity of the union operation is also nearly constant or O(α(n))
+
+// Worst Case Space Complexity: O(n), where n is the number of nodes or element in the structure
+
+// Reference: https://www.scaler.com/topics/data-structures/disjoint-set/
+
+// Author: Mugdha Behere[https://github.com/MugdhaBehere]
+
+// see: unionfind.go, unionfind_test.go
```

Comment:
```suggestion
// Union Find Algorithm or Dynamic Connectivity algorithm, often implemented with the help 
//of the union find data structure,
// is used to efficiently maintain connected components in a graph that undergoes dynamic changes,
// such as edges being added or removed over time
// Worst Case Time Complexity: The time complexity of find operation is nearly constant or 
//O(α(n)), where where α(n) is the inverse Ackermann function
// practically, this is a very slowly growing function making the time complexity for find 
//operation nearly constant.
// The time complexity of the union operation is also nearly constant or O(α(n))
// Worst Case Space Complexity: O(n), where n is the number of nodes or element in the structure
// Reference: https://www.scaler.com/topics/data-structures/disjoint-set/
// Author: Mugdha Behere[https://github.com/MugdhaBehere]
// see: unionfind.go, unionfind_test.go
```

You don't need to add empty lines between comments.